dev-tools/example.sh

```diff
@@ -0,0 +1,399 @@
+
+### Create a record in the scope so that the scope is qualified for onboarding to the contract
+
+RECORD_SPEC_ADDRESS="recspec1q4cgh2qltruylry66e4qwwe99tl5w2alzjfrutnualv99xp9csq7sw828k9"
+RECORD_META_ADDRESS="record1qf3n2cnrxverxvfdxvuk2epdxs6nwvedvg6rgv3dxqckgct9xe3xyvpsv9jkcmmpdcjga2tx"
+
+provenanced tx metadata write-record-specification \
+  "$RECORD_SPEC_ADDRESS" \
+  "loan" \
+  "loanBuilder,org.example.loan.model,someChecksum" \
+  "init,org.example.loan.other,$RECORD_META_ADDRESS" \
+  "record_list" \
+  "owner" \
+  --from "$ORIGINATOR_ACCOUNT" \
+  --testnet \
+  --keyring-backend test \
+  --home "$PROVENANCE_HOME" \
+  --chain-id chain-local \
+  --gas auto \
+  --gas-prices "$GAS_PRICES" \
+  --gas-adjustment "$GAS_ADJUSTMENT" \
+  --yes \
+  --output json | jq '{ txhash }' || exit 1
+
+sleep 3
+
+provenanced query metadata recordspec "$RECORD_SPEC_ADDRESS" --home "$PROVENANCE_HOME" -t -o json | jq >/dev/null 2>&1 && echo "Successfully created a new record specification" || exit 1
+
+provenanced tx metadata write-record \
+  "$SCOPE_ADDRESS" \
+  "$RECORD_SPEC_ADDRESS" \
+  "loan" \
+  "someProcess,anotherChecksum,org.example.loan.model" \
+  "loanBuilder,inputHashValue,org.example.loan.model,proposed" \
+  "newHashValue,pass" \
+  "$ORIGINATOR_ACCOUNT,owner" \
+  "$CONTRACT_SPEC_ADDRESS" \
+  --from "$ORIGINATOR_ACCOUNT" \
+  --testnet \
+  --keyring-backend test \
+  --home "$PROVENANCE_HOME" \
+  --chain-id chain-local \
+  --gas auto \
+  --gas-prices "$GAS_PRICES" \
+  --gas-adjustment "$GAS_ADJUSTMENT" \
+  --yes \
+  --output json | jq '{ txhash }' || exit 1
+
+sleep 3
+
+RECORDS_COUNT=$(
+  provenanced query metadata scope "$SCOPE_ADDRESS" \
+    --include-records \
+    --home "$PROVENANCE_HOME" \
+    --testnet \
+    --output json | jq '.records | length'
+)
+
+if [ -z "$RECORDS_COUNT" ] || [ "$RECORDS_COUNT" -le 0 ]; then
+  echo "Failed to add record to scope"
+  exit 1
+else
+  echo "Successfully added a record to the scope"
+fi
+
+### Onboard the scope to the contract
+
+get_nhash_balance() {
+  provenanced query bank balance "$1" nhash --testnet --home "$PROVENANCE_HOME" -o json | jq -r '.balance.amount'
+}
+
+echo "Loan originator has $(get_nhash_balance "$ORIGINATOR_ACCOUNT") nhash before executing the asset onboarding"
+echo "Loan verifier has $(get_nhash_balance "$VERIFIER_ACCOUNT") nhash before executing the asset onboarding"
+
+provenanced tx wasm execute "$AC_CONTRACT_ADDRESS" \
+  "{
+    \"onboard_asset\": {
+      \"identifier\": {
+        \"type\": \"asset_uuid\",
+        \"value\": \"$SCOPE_UUID\"
+      },
+      \"asset_type\": \"mortgage\",
+      \"verifier_address\": \"$VERIFIER_ACCOUNT\"
+    }
+  }" \
+  --from "$ORIGINATOR_ACCOUNT" \
+  --testnet \
+  --keyring-backend test \
+  --home "$PROVENANCE_HOME" \
+  --chain-id chain-local \
+  --gas auto \
+  --gas-prices "$GAS_PRICES" \
```

Comment:
after changing to `--fees Xnhash` this line can be removed. not sure of an elegant way to auto calculate gas + fees other than scripted